internal/runners/push/push.go

```diff
@@ -187,23 +179,41 @@ func (r *Push) Run(params PushParams) error {
 	return nil
 }
 
-func (r *Push) languageForProject(pj *project.Project) (*language.Supported, string, error) {
-	if pj.CommitID() != "" {
-		lang, err := model.FetchLanguageForCommit(pj.CommitUUID())
+func (r *Push) getNamespace() (*project.Namespaced, error) {
+	namespace := projectfile.GetProjectNameForPath(r.config, r.project.Source().Path())
+	if namespace == "" {
+		owner := authentication.Get().WhoAmI()
+		owner, err := r.prompt.Input("", locale.Tl("push_prompt_owner", "Who will be the owner of this project?"), &owner)
 		if err != nil {
-			return nil, "", errs.Wrap(err, "Failed to retrieve language information for headless commit.")
+			return nil, locale.WrapError(err, "err_push_get_owner", "Could not deterimine project owner")
 		}
 
-		l, err := language.MakeByNameAndVersion(lang.Name, lang.Version)
+		lang, _, err := fetchLanguage(r.project.CommitUUID())
 		if err != nil {
-			return nil, "", errs.Wrap(err, "Failed to convert commit language to supported language.")
+			return nil, locale.WrapError(err, "err_push_fetch_language", "Could not fetch project language")
 		}
```

Comment:
This shouldn't actually cause a failure if it fails, we should just fall back on an empty name.